modules/sox-workflowoperation/src/main/java/org/opencastproject/workflow/handler/sox/AnalyzeAudioWorkflowOperationHandler.java

```diff
@@ -118,43 +118,32 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, Jo
     logger.debug("Running analyze audio workflow operation on workflow {}", workflowInstance.getId());
 
     try {
-      return analyze(workflowInstance.getMediaPackage(), workflowInstance.getCurrentOperation());
+      return analyze(workflowInstance.getMediaPackage(), workflowInstance.getCurrentOperation(), workflowInstance);
     } catch (Exception e) {
       throw new WorkflowOperationException(e);
     }
   }
 
-  private WorkflowOperationResult analyze(MediaPackage src, WorkflowOperationInstance operation) throws SoxException,
+  private WorkflowOperationResult analyze(MediaPackage src, WorkflowOperationInstance operation, WorkflowInstance wi) throws SoxException,
           IOException, NotFoundException, MediaPackageException, WorkflowOperationException, EncoderException {
     MediaPackage mediaPackage = (MediaPackage) src.clone();
 
-    // Check which tags have been configured
-    String sourceTagsOption = StringUtils.trimToNull(operation.getConfiguration("source-tags"));
-    String sourceFlavorOption = StringUtils.trimToNull(operation.getConfiguration("source-flavor"));
-    String sourceFlavorsOption = StringUtils.trimToNull(operation.getConfiguration("source-flavors"));
+    ConfiguredTagsAndFlavors tagsAndFlavors = getTagsAndFlavors(wi, Configuration.many, Configuration.many, Configuration.none, Configuration.none);
+
     boolean forceTranscode = BooleanUtils.toBoolean(operation.getConfiguration("force-transcode"));
 
     AbstractMediaPackageElementSelector<Track> elementSelector = new TrackSelector();
 
     // Make sure either one of tags or flavors are provided
-    if (StringUtils.isBlank(sourceTagsOption) && StringUtils.isBlank(sourceFlavorOption)
-            && StringUtils.isBlank(sourceFlavorsOption)) {
+    List<String> sourceFlavorsOption = tagsAndFlavors.getSrcFlavors();
```

Comment:
Why call this option? These are now the source flavors, aren't they?

```suggestion
    List<String> sourceFlavors = tagsAndFlavors.getSrcFlavors();
```

(Same with tags)